remap: panic on source osd missing from up set

When the cluster is not in a happy state, and the source OSD is not in the up set, `pgremapper remap` will panic, preventing the user from trying to reconstruct data from a down OSD on a specific target.

Some years ago, `pgremapper` have many of panics, see https://github.com/digitalocean/pgremapper/issues/10#issuecomment-945648879
Today, the master code works for us in every-day usage. So, if you think you find the new bug, lease read, how to [Bug Reports](https://github.com/digitalocean/pgremapper?tab=readme-ov-file#bug-reports)

For a quick fix you can use this shell function

```bash
function fix_pgremapper_panic() {
  # function fix situations when pgremapper is crashed on some osdmaps
  local pg
  local upmap_cmd
  upmap_cmd="--verbose"

  if [[ -n "${UPMAP_POOL}" ]]
    then
      upmap_cmd="${upmap_cmd} --include-pools ${UPMAP_POOL}"
  fi

  # shellcheck disable=SC2086
  until ${PG_REMAPPER} cancel-backfill ${upmap_cmd} > /dev/null 2>&1
    do
      pg="$(${PG_REMAPPER} cancel-backfill ${upmap_cmd} 2>&1 >/dev/null | \
        grep "panic" | tr -d '\n' | awk '{print $3}' | sed 's|:||g')"
      echo -e "Fix pgremapper panic for PG ${pg}"
      ceph osd rm-pg-upmap-items "${pg}"
      sleep 1s
    done
}
```

--- a/backfillstate.go
+++ b/backfillstate.go
@@ -77,25 +77,22 @@ func (bs *backfillState) accountForRemap(pgid string, from, to int) {
 	if !ok {
 		panic(fmt.Sprintf("%s: no such PG", pgid))
 	}
-	bs.removeReservations(pgb)
 
-	found := false
 	for i, osd := range pgb.Up {
 		if osd == from {
-			found = true
+			bs.removeReservations(pgb)
 			pgb.Up[i] = to
-			break
+			// Do not use the upmap here as we don't need to strictly re-order the
+			// up set; it's sufficient to consider which OSDs are listed in up and
+			// acting by themselves.
+			reorderUpToMatchActing(pgid, pgb.Up, pgb.Acting, false)
+			bs.addReservations(pgb)
 		}
 	}
-	if !found {
-		panic(fmt.Sprintf("%s: osd %d not in up set", pgid, from))
-	}
-	// Do not use the upmap here as we don't need to strictly re-order the
-	// up set; it's sufficient to consider which OSDs are listed in up and
-	// acting by themselves.
-	reorderUpToMatchActing(pgid, pgb.Up, pgb.Acting, false)
-
-	bs.addReservations(pgb)
+	// We can get here if a remap has been requested where the 'from' OSD
+	// is currently down. As noted in the osdBackfillState type TODO, we
+	// don't handle degraded backfill today.
+	fmt.Printf("pg %s: osd %d not in up set, unable to compute effect of remap on backfill state", pgid, from)
 }
 
 func (bs *backfillState) addReservations(pgb *pgBriefItem) {

